test/export/InstanceExporter.test.ts

```diff
@@ -247,5 +247,28 @@ describe('InstanceExporter', () => {
       instance.rules.push(instanceFixedValRule);
       expect(() => exporter.exportInstance(instance)).toThrow();
     });
+
+    // Fixing children of primitives
+    it('should fix children of primitive values on an instance', () => {
+      const fixedValRule = new FixedValueRule('active.id');
+      fixedValRule.fixedValue = 'foo';
+      instance.rules.push(fixedValRule);
+      doc.instances.set(instance.name, instance);
+      const exported = exporter.exportInstance(instance);
+      expect(exported._active.id).toBe('foo');
+    });
+
+    it('should fix children of primitive value arrays on an instance', () => {
+      const fixedValRule = new FixedValueRule('address[0].line[1].extension[0].url');
+      fixedValRule.fixedValue = 'foo';
+      instance.rules.push(fixedValRule);
+      doc.instances.set(instance.name, instance);
+      const exported = exporter.exportInstance(instance);
+      expect(exported.address.length).toBe(1);
+      expect(exported.address[0]._line.length).toBe(2);
+      expect(exported.address[0]._line[0]).toBeNull();
+      expect(exported.address[0]._line[1].extension.length).toBe(1);
+      expect(exported.address[0]._line[1].extension[0].url).toBe('foo');
+    });
```

Comment:
I found an issue when taking a look at this PR. It might be more appropriately addressed in a separate task, as it isn't introduced by your changes. It has to do with setting values on array elements when the indices don't arrive in ascending order. If the following block as line 264, a `TypeError` occurs in `Object.setPropertyOnInstance (src/fhirtypes/common.ts:43:20)` when attempting to apply the rule on `address[0].line[1].extension[0].url` because `line[1]` exists, but is null.
```
      const badStuff = new FixedValueRule('address[0].line[2].extension[0].url');
      badStuff.fixedValue = 'bogus';
      instance.rules.push(badStuff);
```

Reply:
Good find, @mint-thompson.  @ngfreiter -- let's make a unit test for this situation as well.